src/storm/generator/PrismNextStateGenerator.cpp

```diff
@@ -592,6 +593,22 @@ PrismNextStateGenerator<ValueType, StateType>::getActiveCommandsByActionIndex(ui
     return result;
 }
 
+template<typename ValueType, typename BaseValueType>
+ValueType evaluateLikelihoodExpression(storm::prism::Update const& update, storm::expressions::ExpressionEvaluator<BaseValueType> const& evaluator) {
+    if constexpr (storm::IsIntervalType<ValueType>) {
+        if (update.isLikelihoodInterval()) {
+            BaseValueType lower = evaluator.asRational(update.getLikelihoodExpressionInterval().first);
+            BaseValueType upper = evaluator.asRational(update.getLikelihoodExpressionInterval().second);
+            return ValueType(lower, carl::BoundType::WEAK, upper, carl::BoundType::WEAK);
```

Comment:
An assertion/check with -ec on the lower bound being at least zero and the upper bound being at most one would probably be helpful (although one can argue where to do this?)

Reply:
I added checks for `0 <= probability <= 1` at places where `evaluateLikelihoodExpression` is called. This is also usefull for non-interval updates (although we also have checks that the sum of all probabilities is equal to 1).

However, I wonder if this actually needs to be enforced with intervals. 
One could argue that `[-1,2]` is a valid probability interval (and equivalent to `[0,1]`). Thoughts?

Reply:
I think the problem is if you have multiple parallel components, then, because [-2, 1] * [-2, 1] = [-2, 4], which is not what you would get. We don't get these problems if we ensure that we do not allow negative probabilities in our intervals.